tsinfer/constants.py

```diff
@@ -37,6 +37,9 @@
 # file
 NODE_IS_HISTORICAL_SAMPLE = 1 << 20
 
+# Undocumented.
+NODE_IS_IDENTICAL_SAMPLE_ANCESTOR = 1 << 21
```

Comment:
I think we call these "proxy ancestors" in other code (i.e. an ancestor identical to a sample inserted  a little earlier in time, to help matching)